packages/web3-core-promievent/src/index.js

```diff
@@ -23,7 +23,6 @@
 "use strict";
 
 var EventEmitter = require('eventemitter3');
-var Promise = require("any-promise");
```

Comment:
Is this a required change or a 'nice-to-have'? The 'any-promise' fix was made in #1415 and the PR is not super clear about implications. If you left it as it is would that be bad?

Reply:
Promise will be polyfilled by our build pipeline the crypto dependency also will get polyfilled for the browser with the browserify-crypto package. (I've checked that closer)

Reply:
If this isn't absolutely necessary, can we please put it in a separate PR later? I'm seeing failing tests at Truffle around the promi-event package and this change will make it harder to tell exactly where the problem is (e.g here or at Truffle) when these changes are run against that E2E.